File: 0739-daily-temperatures/0739-daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        stack: List = [] # (idx,t)
        res: List = [0 for t in (temperatures)]

        for (idx,temp) in enumerate(temperatures):
            while (len(stack)>0 and stack[-1][1] < temp ):
                (stack_idx, _) = stack.pop()
                dist = idx - stack_idx
                res[stack_idx] = dist

            stack.append((idx,temp))
        return res

File: 0739-daily-temperatures/0739-daily-temperatures.ts
type T = {
  temp: number;
  idx: number;
}

function dailyTemperatures(temperatures: number[]): number[] {
  let stack: T[] = [];
  let res: number[] = new Array(temperatures.length).fill(0);

  for (let i=0; i<temperatures.length; i++){
    let cur_temp: number = temperatures[i];
    while ((stack.length > 0) && (stack[stack.length -1].temp < cur_temp) ){
      let { idx } = stack.pop();
      res[idx] = i - idx;
    }
    stack.push({ temp: cur_temp, idx:i})
  }

  return res;
};
